pri/boostdetect.pri

```diff
@@ -13,44 +13,55 @@
 # along with Fritzing. If not, see <http://www.gnu.org/licenses/>.
 # ********************************************************************/
 
+LATESTBOOST = 0
 
-exists($$boost_root) {
-    INCLUDEPATH += $$absolute_path($$boost_root)
-    message("using boost $$absolute_path($$boost_root)")
+# reduce warning messages by checking if variables are defined before using them
+defined(boost_root, var) {
+    LATESTBOOST = user
+    exists($$boost_root) {
+        # force absolute: relative would break due to detection versus use folder nest level
+        BOOSTPATH = $$absolute_path($$boost_root)
+        absolute_boost = 1
+        unset(absolute_boost)
+    } else {
+        error("requested boost_root $$boost_root does not exist")
+    }
 } else {
-# exists($$BOOST_ROOT) {
-#     INCLUDEPATH += $$BOOST_ROOT
-#     message("using BOOST_ROOT environment")
-# } else {
     # message("Boost auto detect is deprecated, please set boost root variable.")
     message("Using fritzing boost detect script.")
 
-
     # boost_1_54_0 is buggy
     BOOSTS = 43 44 45 46 47 48 49 50 51 52 53 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99
-    LATESTBOOST = 0
     for(boost, BOOSTS) {
         exists(../../boost_1_$${boost}_0) {
             LATESTBOOST = $$boost
+            BOOSTPATH = ../boost_1_$${boost}_0
         }
         exists(../src/lib/boost_1_$${boost}_0) {
             LATESTBOOST = $$boost
+            BOOSTPATH = src/lib/boost_1_$${boost}_0
```

Comment:
What do you think about removing that magic location? I kept it until now, to avoid breaking old documentation or existing build environments, but I don't really think its a good idea to copy boost libs into the fritzing folder.

Reply:
I think it would be a good idea. With the extra *intent* information from your earlier comment.

Reply:
It would still be **possible** to use that path (after removal of the code block), by using the boost_root environment variable. Can't stop people from shooting themselves.